pac4j-core/src/main/java/org/pac4j/core/client/BaseClient.java

```diff
@@ -56,11 +57,11 @@
      * @param context the web context
      * @return the credentials
      */
-    protected C retrieveCredentials(final WebContext context) {
+    protected Optional<C> retrieveCredentials(final WebContext context) {
         try {
             final C credentials = this.credentialsExtractor.extract(context);
```

Comment:
Why not make credentialsExtractor return an optional?

Reply:
In fact this change should be applied to `ProfileCreator` and `AuthorizationGenerator`